src/layout-rect.js

```diff
@@ -77,23 +77,36 @@ export function layoutRectsOverlap(r1, r2) {
 
 /**
  * Returns the intersection between a, b or null if there is none.
- * @param {!LayoutRectDef} a
- * @param {!LayoutRectDef} b
+ * @param {!Array<?LayoutRectDef|undefined>} layoutRects
  * @return {?LayoutRectDef}
  */
-export function rectIntersection(a, b) {
-  const x0 = Math.max(a.left, b.left);
-  const x1 = Math.min(a.left + a.width, b.left + b.width);
+export function rectIntersection(layoutRects) {
```

Comment:
https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#variable-parameters-in-param-annotations